src/map.cpp

```diff
@@ -1180,6 +1180,14 @@ bool Map::isBlockOccluded(MapBlock *block, v3s16 cam_pos_nodes)
 	// this is a HACK, we should think of a more precise algorithm
 	u32 needed_count = 2;
 
+	v3s16 random_point(myrand_range(-bs2, bs2), myrand_range(-bs2, bs2), myrand_range(-bs2, bs2));
+	if (!isOccluded(cam_pos_nodes, pos_blockcenter + random_point, step, stepfac,
```

Comment:
this adds an additional occlusion check to every call, did you mean to put this into the `if`?

Reply:
I actually meant it like this. I realized that an extra random probe would improve occlusion culling overall over time, over the course of multiple sweeps it will get more accurate even if we do the 9 other probes.

For smaller distances it makes no difference (performance wise), and the for larger distances it is the only check.